app/controllers/queue_items_controller.rb

```diff
@@ -14,13 +14,24 @@ def create
   def destroy
     queue_item = QueueItem.find(params[:id])
     queue_item.destroy if current_user.queue_items.include?(queue_item)
+    current_user.normalize_queue_item_positions
     redirect_to my_queue_path
   end
 
+  def update_queue
+   begin
+     update_queue_items
+     current_user.normalize_queue_item_positions
+   rescue ActiveRecord::RecordInvalid
+      flash[:error] = "Invalid position number."
+   end
+   redirect_to my_queue_path    
```

Comment:
Lines 22, 23, 24, 25, 27, 28 should be 1 more space to right :) its wrongly indented now.